docker-compose.yml

```diff
@@ -0,0 +1,33 @@
+version: "3.8"
+
+services:
+  mongo1:
+    image: mongo:5
+    container_name: mongo1
+    command: ["--replSet", "my-replica-set", "--bind_ip_all", "--port", "30001"]
```

Comment:
does default configuration support replica set?
where to config the database if we want to use single node mongo db?

Reply:
here is the story.
this is my first time use Prisma orm, I know it is bit risk use new lib/framework which I don't familiar during interview project. But I always want to learn something new from each interview. I spent few hours read the doc and I found it is only supports replica. Also I think it is generally considered a good practice using MongoDB in a replica set configuration.
and if we really need use single node mongodb, we can easily replaced it with other orm framework, that is why I use interface injected to the userService instead of inject concrete prismaService class. the code can be much more simplified if I directly inject prismaService to userService but I think it lots the flexibility  and  maintainability.

Reply:
it's ok to use replica set, does the default configuration support replica set?
where to configure it if another environment uses more nodes or other ports?